### PR TITLE
Site Migration: add ability to store and restore progress

### PR DESCRIPTION
**Pull request by heavyweight**
This PR stores the URL to local storage whenever the user progresses through the site migration flow.
In our email campaigns later we can leverage this to restore the progress of the user in the flow.

Related to #
pdDR7T-1sI-p2

## Proposed Changes

* Store the URL when site migration step changes
* Retrieve the stored URL and redirect when the `restore-progress` URL parameter is present 

## Testing Instructions
- Build the PR or use the image link
- In incognito window follow this flow /setup/import-hosted-site?ref=hosting-lp&section=performance
- Create a new site on tastewp
- Input that URL in the first step
- Go to the next step, check in the local storage if the URL is stored
- Open a new tab and visit: /setup/import-hosted-site?restore-progress
- The progress should be restored and you should be redirected to your last step

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by agrullon95** [comment]
The logic is working well. I'm able to navigate back and forth through the site migration step and have the URL stored in local storage, then open ` /setup/site-migration/?siteSlug=YOURSITEURL&restore-progress` in another tab and Im navigated back to the last step I was on.

From a visual perspective, I do sometimes the "URL input" step flashing for a moment.

Also, this logic will apply for all users on the same computer, so they will be navigated to the same step if the `restore-progress` param exists. I think that scenario is very unlikely though.

**Review by escapemanuele — CHANGES_REQUESTED** [review]
Thanks for working on this!

I'll check where we want to redirect the users after they click on the email CTA

File: client/landing/stepper/declarative-flow/site-migration-flow.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ const siteMigration: Flow = {
 			STEPS.ERROR,
 		];
 	},
+
```
Comment (escapemanuele) [inline client/landing/stepper/declarative-flow/site-migration-flow.ts]: Probably better to not touch this flow in this PR

**Review by escapemanuele — APPROVED** [review]
Works great for me now! 👍

**Pull request merged**